jenkins/README.md

```diff
@@ -0,0 +1,100 @@
+# Jenkins
+
+[Jenkins](http://jenkins-ci.org/) is a pluggable continuous
+integration system. The Google team is running two Jenkins servers in GCE for
+the Kubernetes project. The post-commit instance runs continuous builds, unit
+tests, integration tests, code verification tests, and end-to-end tests on
+multiple providers using the latest commits to the Kubernetes repo from the
+master and release branches. The PR Jenkins instance runs these tests on each
+PR by a trusted contributor, it but only runs a subset of the end-to-end tests
+and only on GCE.
+
+## General flow
+The flow of the post-commit Jenkins instance:
+* Under the `kubernetes-build` job: Every 2 minutes, Jenkins polls for a batch
+  of new commits, after which it runs the `build.sh` script (in this directory)
+  on the latest tip. This results in build assets getting pushed to GCS and the
+  `latest.txt` file in the `ci` bucket being updated.
+* On trigger, and every half hour (which effectively means all the time, unless
+  we're failing cluster creation), e2e variants run, on the latest build assets
+  in GCS:
+  * `kubernetes-e2e-gce`: Standard GCE e2e.
+  * `kubernetes-e2e-gke`: GKE provider e2e, with head k8s client and GKE
+    creating clusters at its default version.
+  * `kubernetes-e2e-aws`: AWS provider e2e. This only runs once a day.
+* Each job will not run concurrently with itself, so, for instance,
+  Jenkins executor will only ever run one `kubernetes-build`
+  job. However, it may run the jobs in parallel,
+  i.e. `kubernetes-build` may be run at the same time as
+  `kubernetes-e2e-gce`. For this reason, you may see your changes
+  pushed to our GCS bucket rapidly, but they may take some time to
+  fully work through Jenkins. Or you may get lucky and catch the
+  train in 5 minutes.
+* There are many jobs not listed here, including upgrade tests, soak tests, and
+  tests for previous releases.
+
+## Scripts
+
+The scripts in this directory are directly used by Jenkins, either by
```

Comment:
I'd like to move the scripts over eventually, but for now they aren't actually in this directory.